distribution/lib/Standard/Base/0.0.0-dev/src/Data/Range.enso

```diff
@@ -249,8 +249,9 @@ type Range
             end_condition = if self.step > 0 then (>=) else (<=)
             go current =
                 if end_condition current self.end then Nothing else
-                    function current
-                    @Tail_Call go current+self.step
+                    result = function current
+                    if result.is_error then Error.throw result else
+                        @Tail_Call go current+self.step
```

Comment:
With the idiom it can even be shortened avoiding the intermediate assignment:
```suggestion
                    function current . if_not_error <|
                        @Tail_Call go current+self.step
```

Reply:
Fixed